stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1701,6 +1707,347 @@ fn delegate_stack_stx_signer_key() {
+        stacker_nonce,
+        "delegate-stack-extend",
+        vec![
+            PrincipalData::from(key_to_stacks_addr(stacker_key)).into(),
+            pox_addr.clone(),
+            signer_key_new_val.clone(),
+            Value::UInt(1),
+        ],
+    )];
+
+    latest_block = peer.tenure_with_txs(&update_txs, &mut coinbase_nonce);
+    let new_stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .unwrap()
+    .expect_tuple();
+
+    let state_signer_key_new = new_stacking_state.get("signer-key").unwrap();
+    assert_eq!(
+        state_signer_key_new.to_string(),
+        signer_key_new_val.to_string()
+    );
+}
+
+#[test]
+fn stack_increase() {
+    let lock_period = 2;
+    let observer = TestEventObserver::new();
+    let (burnchain, mut peer, keys, latest_block, block_height, mut coinbase_nonce) =
+        prepare_pox4_test(function_name!(), Some(&observer));
+
+    let mut stacker_nonce = 0;
+    let stacker_key = &keys[0];
+    let stacker_address = key_to_stacks_addr(stacker_key);
+    let min_ustx = get_stacking_minimum(&mut peer, &latest_block);
+
+    // (define-public (stack-stx (amount-ustx uint)
+    //                       (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+    //                       (start-burn-ht uint)
+    //                       (lock-period uint)
+    //                       (signer-key (buff 33)))
+    let pox_addr = make_pox_addr(
+        AddressHashMode::SerializeP2WSH,
+        key_to_stacks_addr(stacker_key).bytes,
+    );
+    let signer_key_val = Value::buff_from(vec![
+        0x03, 0xa0, 0xf9, 0x81, 0x8e, 0xa8, 0xc1, 0x4a, 0x82, 0x7b, 0xb1, 0x44, 0xae, 0xc9, 0xcf,
+        0xba, 0xeb, 0xa2, 0x25, 0xaf, 0x22, 0xbe, 0x18, 0xed, 0x78, 0xa2, 0xf2, 0x98, 0x10, 0x6f,
+        0x4e, 0x28, 0x1b,
+    ])
+    .unwrap();
+    let first_txs = vec![make_pox_4_contract_call(
+        stacker_key,
+        stacker_nonce,
+        "stack-stx",
+        vec![
+            Value::UInt(min_ustx),
+            pox_addr,
+            Value::UInt(block_height as u128),
+            Value::UInt(2),
+            signer_key_val.clone(),
+        ],
+    )];
+
+    let latest_block = peer.tenure_with_txs(&first_txs, &mut coinbase_nonce);
+    let stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(stacker_key).to_account_principal(),
+    )
+    .expect("No stacking state, stack-stx failed")
+    .expect_tuple();
+
+    let state_signer_key = stacking_state.get("signer-key").unwrap();
+    assert_eq!(state_signer_key.to_string(), signer_key_val.to_string());
+
+    stacker_nonce += 1;
+
+    // (define-public (stack-increase (increse-by uint)
+    let stack_increase = make_pox_4_increase_stx(stacker_key, stacker_nonce, min_ustx);
+    let latest_block = peer.tenure_with_txs(&vec![stack_increase], &mut coinbase_nonce);
```

Comment:
at other places you create a tx array and use it as an argument, here you create the array in the argument. Better do the further.